lib/socket.js

```diff
@@ -21,19 +21,26 @@ function Socket(controller) {
   this._neverReceived = true;
   this._neverSent = true;
   this._waitingAsync = false;
+  this.dataFragment = '';  // Prevents partial packets from getting lost (added by heberallred)
 }
 
 Socket.prototype._onConnect = function () {
   this._connected = true;
   this._controller.emit('connected');
 
-  this._controller.run('sendAsync', [C.CLIENT_VERSION]);
-  this._controller.run('sendAsync', [this._controller.options.clientId]);
+  this._controller.run('sendAsync', [C.CLIENT_VERSION, this._controller.options.clientId]);
```

Comment:
@lewisdawson I'm not sure why these were combined, but let's break them back out into two calls again and see what happens